widgetschema.json

```diff
@@ -228,8 +233,53 @@
                             "items": { "$ref": "#/definitions/dataType" }
                         }
                     }
+                },
+                "visibility": { "$ref": "#/definitions/WidgetVisibilityGroup" },
+                "fields": {
+                    "type": "array",
+                    "items": { "$ref": "#/definitions/widgetProperty" }
+                },
+                "itemName": {
```

Comment:
I'm assuming `fields` are the properties of the repeated nested objects inside a "multiple" property, but I'm not following what `itemName` is.

Also, do `fields` and `itemName` have any meaning outside of a "multiple" property?  If not, should these be defined in the "multiple" definition above in `oneOf`?

Reply:
I see.  So a multiple property might have a plural form like "Buttons", and then `itemName` would have the singular form "Button"?

One concern is that `itemName` might be seen as similar to `name`, i.e. a machine-readable id, but really it seems to be more like `displayName`, a label for presentation purposes only.  I also have trouble reconciling that `displayName` is optional but `itemName` is required.  I would think they're both optional.